src/main/java/org/broadinstitute/dsm/util/ElasticSearchUtil.java

```diff
@@ -1012,6 +1033,23 @@ else if (nameValue[0].startsWith(INVITATIONS)) {
                                 catch (ParseException e) {
                                     logger.error("range was not date. user entered: " + userEntered);
                                 }
+                            } else if (StringUtils.isNumeric(userEntered)) { // separately process expressions with numbers
+                                QueryBuilder tmpBuilder = findQueryBuilderForFieldName(finalQuery, surveyParam[1], queryPartsMap);
```

Comment:
I think it will be good idea to extract content of this if block as a method, since nearly the very same code is written above, from L933 - L949, to avoid code duplication.

Reply:
Extracted the code.